internals/src/error.rs

```diff
@@ -27,3 +27,15 @@ macro_rules! write_err {
         }
     }
 }
+
+/// Trait used internally to assert that error types implement `Send`, `Sync`, and `Debug`.
+#[doc(hidden)]
+pub trait AssertSendSyncDebug: Send + Sync + core::fmt::Debug {}
```

Comment:
Why is this pub and why do we need a trait? Any drawbacks of creating this in `#cfg(test)` for this? We can also do this with function without the trait. 

```rust
#[cfg(test)
fn assert_send_sync<T: Send + Sync + Debug> {}

macro_rules check_pub_error {
	#[cfg(test)]
	assert_send_sync<$type>();	
}
```

Reply:
Could be `pub(crate)` I think.

Reply:
I ended up using a trait still but feature guarding so the trait is not built in in non-test builds. Ok?

Reply:
Sure. I am okay as long it is only in tests.